### PR TITLE
chore: Add more e2e Sub-Workflow Input testing

### PR DESCRIPTION
**Pull request by CharlieKolb**
## Summary

Test json mode and add WIP test for type option flags and resourceMapper fields.

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/ADO-2915/add-tests-for-the-feature

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: cypress/e2e/48-subworkflow-inputs.cy.ts [hunk 1]
```diff
@@ -83,6 +93,21 @@ function navigateWorkflowSelectionDropdown(index: number, expectedText: string)
 		.click();
 }
 
+function populateMapperFields(values: readonly string[], offset: number) {
```
Comment (MiloradFilipovic) [inline cypress/e2e/48-subworkflow-inputs.cy.ts:96]: Does it make sense to move these local functions to `actions` (maybe [here](https://github.com/n8n-io/n8n/blob/master/cypress/pages/ndv.ts))? I know these pages are getting out of hand (we can maybe create a new one just for RMC), so feel free to push back, but I am worried about mixing actions and utli functions to do the same thing.

**Review by MiloradFilipovic — APPROVED** [review]
Looks, good. Left a comment to consider.

File: cypress/e2e/48-subworkflow-inputs.cy.ts (hunk 1 above)
Comment (CharlieKolb) [reply cypress/e2e/48-subworkflow-inputs.cy.ts:96]: Yes, strongly agree with this needing a refactor before (maybe master) merge, I'm defining them as I'm iterating for now. I'm also not happy with the sub-functions I'm defining in the big test case.

Comment (MiloradFilipovic) [reply cypress/e2e/48-subworkflow-inputs.cy.ts:96]: Yes, we have it in few other places, it's not a bad pattern to be honest it's just about consistency. But, looks like a big push to get e2e tests in order is around the corner so good to go like this

**Pull request merged**